tests/Microsoft.Identity.Test.LabInfrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -51,7 +51,8 @@ public class KeyVaultSecretsProvider : IDisposable
 
         private KeyVaultConfiguration _config;
 
-        private readonly string _keyVaultClientID = "ebe49c8f-61de-4357-9194-7a786f6402b4";
+        private readonly string _keyVaultConfidentialClientID = "ebe49c8f-61de-4357-9194-7a786f6402b4";
+        private readonly string _keyVaultPublicClientID = "3c1e0e0d-b742-45ba-a35e-01c664e14b16";
```

Comment:
Interesting, so KV defines separate apps for public access and confidential access?

Reply:
Not exactly. The public client application should only be used with applications marked as "native" in the azure portal. The app used by the confidential client is "web app/api".